Sources/Controllers/2.0/OAEditColorViewController.m

```diff
@@ -86,27 +88,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 {
     static NSString* const reusableIdentifierPoint = @"OAViewTextTableViewCell";
     
-    OAViewTextTableViewCell* cell;
-    cell = (OAViewTextTableViewCell *)[self.tableView dequeueReusableCellWithIdentifier:reusableIdentifierPoint];
+    OAIconTextTableViewCell* cell;
+    cell = (OAIconTextTableViewCell *)[self.tableView dequeueReusableCellWithIdentifier:reusableIdentifierPoint];
     if (cell == nil)
     {
-        NSArray *nib = [[NSBundle mainBundle] loadNibNamed:@"OAViewTextCell" owner:self options:nil];
-        cell = (OAViewTextTableViewCell *)[nib objectAtIndex:0];
+        NSArray *nib = [[NSBundle mainBundle] loadNibNamed:kCellTypeCheck owner:self options:nil];
+        cell = (OAIconTextTableViewCell *)[nib objectAtIndex:0];
     }
     
     if (cell) {
         
         OAFavoriteColor *favCol = [OADefaultFavorite builtinColors][indexPath.row];
         [cell.textView setText:favCol.name];
-        [cell.titleIcon setImage:favCol.icon];
-        [cell.iconView setImage:nil];
-        
-        if (indexPath.row == self.colorIndex)
-            [cell.iconView setImage:[UIImage imageNamed:@"menu_cell_selected"]];
+        [cell.iconView setImage:favCol.icon];
+        //[cell.iconView setImage:nil];
+        cell.textLeftMargin.constant -= 8.;
+        [cell.arrowIconView setImage:[UIImage imageNamed:@"menu_cell_selected"]];
+        cell.arrowIconView.hidden = indexPath.row != self.colorIndex;
     }
     return cell;
 }
 
+- (CGFloat) tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    return UITableViewAutomaticDimension;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    return UITableViewAutomaticDimension;
```

Comment:
estimatedHeightForRowAtIndexPath should not be automatic + both this methods are redundant. You should set the values in the xib and remove them